addons/mail/models/mail_message.py

```diff
@@ -997,31 +997,18 @@ def _to_store(
                 record_name = False
                 default_subject = False
             data["default_subject"] = default_subject
-            reactions_per_content = defaultdict(self.env["mail.message.reaction"].sudo().browse)
             # sudo: mail.message - reading reactions on accessible message is allowed
             reactions = message.sudo().reaction_ids
-            for reaction in reactions:
-                reactions_per_content[reaction.content] |= reaction
             store.add(reactions.guest_id, fields=["name", "write_date"])
             store.add(reactions.partner_id, fields=["name", "write_date"])
-            reaction_groups = [
-                {
-                    "content": content,
-                    "count": len(reactions),
-                    "personas": Store.many_ids(reactions.guest_id)
-                    + Store.many_ids(reactions.partner_id),
-                    "message": Store.one_id(message),
-                }
-                for content, reactions in reactions_per_content.items()
-            ]
             vals = {
                 # sudo: mail.message - reading attachments on accessible message is allowed
                 "attachments": Store.many(message.sudo().attachment_ids.sorted("id")),
                 # sudo: mail.message - reading link preview on accessible message is allowed
                 "linkPreviews": Store.many(
                     message.sudo().link_preview_ids.filtered(lambda l: not l.is_hidden)
                 ),
-                "reactions": reaction_groups,
+                "reactions": reactions._reaction_groups().get(message.id, []),
```

Comment:
This should be using `Store.many`

Reply:
Note: you'll need to add special case for reaction group in `one_id` to generate the proper id when using `many`.